compiler/src/pp_arm_m4.ml

```diff
@@ -33,8 +33,28 @@ let pp_reg_address_aux base disp off scal =
   | _, _, _ ->
       failwith "TODO_ARM: pp_reg_address_aux"
 
-let pp_rip_address (_ : Ssralg.GRing.ComRing.sort) : string =
-  failwith "TODO_ARM: pp_rip_address"
+(* TODO_ARM: get rid of global state *)
+let label_of_global_offset, all_global_offsets =
+  let fresh =
+    let counter = ref (-1) in
+    fun () ->
+      incr counter;
+      !counter
+  in
+  let m = Hashtbl.create 137 in
+  ( (fun z ->
+      match Hashtbl.find m z with
+      | n -> n
+      | exception Not_found ->
+          let n = Format.asprintf ".jasmin_global_%d" (fresh ()) in
+          Hashtbl.add m z n;
+          n),
+    fun () ->
+      Hashtbl.fold (fun z n a -> (z, n) :: a) m []
```

Comment:
Is this different from `Hashtbl.to_list`?

Reply:
Thanks. I was looking in the implementation from the stdlib…